gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -88,21 +87,35 @@ public void mapObservationToImmunizationWhenEhrCompositionIdIsNotPresentOnEncoun
     @Test
     public void mapObservationToImmunizationWithMultipleObservationStatements() {
         var ehrExtract = unmarshallEhrExtract("full_valid_immunization_with_multiple_observation_statements.xml");
-        List<Immunization> immunizationList = immunizationMapper.mapResources(ehrExtract, getPatient(), getEncounterList(),
-            PRACTISE_CODE);
+        List<Immunization> immunizationList = immunizationMapper.mapResources(ehrExtract, getPatient(), getEncounterList(), PRACTISE_CODE);
 
         var immunization = (Immunization) immunizationList.get(0);
-        assertThat(immunizationList.size()).isEqualTo(THREE);
+        assertThat(immunizationList).hasSize(THREE);
         assertThat(immunization.getId()).isEqualTo(OBSERVATION_ROOT_ID);
         assertThat(immunization.getMeta().getProfile().get(0).getValue()).isEqualTo(META_PROFILE);
         assertThatIdentifierIsValid(immunization.getIdentifierFirstRep(), immunization.getId());
     }
 
+    @Test
+    public void mapObservationToImmunizationWithMultipleObservationStatementsAndCodingDisplay() {
+        final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-VaccinationProcedure-1";
+        var ehrExtract = unmarshallEhrExtract("full_valid_immunization_with_multiple_observation_statements.xml");
+        List<Immunization> immunizationList = immunizationMapper.mapResources(ehrExtract, getPatient(), getEncounterList(), PRACTISE_CODE);
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertThat(immunizationList).hasSize(THREE);
```

Comment:
I don't know about anybody else in the team, but I'm a fan of the Single Assert rule.

In other words, this test should only be checking for one behaviour.

At the moment, we've got several asserts and from the test name I'm not clear what this test should be specifically checking for.

Reply:
yes, agree that the test is cluttered with other not-relevant test asserts hence I adjusted it.